lib/active_admin/views/components/active_admin_form.rb

```diff
@@ -131,12 +131,22 @@ def build(form_builder, *args, &block)
         html_options[:class] ||= "inputs"
         legend = args.shift if args.first.is_a?(::String)
         legend = html_options.delete(:name) if html_options.key?(:name)
-        legend_tag = legend ? "<legend class=\"fieldset-title\">#{ERB::Util.html_escape(legend)}</legend>" : ""
-        fieldset_attrs = html_options.map { |k, v| %Q{#{k}="#{v}"} }.join(" ")
+        legend_tag = legend ? tag.legend(legend, class: "fieldset-title") : ""
+        fieldset_attrs = tag_attributes html_options
         @opening_tag = "<fieldset #{fieldset_attrs}>#{legend_tag}<ol>"
         @closing_tag = "</ol></fieldset>"
         super(*(args << html_options), &block)
       end
+
+      private
+
+      def tag_attributes(html_options)
+        # When support for Rails 6.x is dropped, use tag.attributes instead.
+        #
+        #     tag.attributes html_options
+        #
+        tag.tag_options(html_options.to_h).to_s.strip.html_safe
+      end
```

Comment:
If I understand right this is done to support both Rails 6.1 and 7+ but I'd prefer that we use a condition so if it's Rails 6.1 we use the `tag.tag_options` method, otherwise always use the current `tag.attributes` method instead. I would expect the former to be removed at some point in Rails so that's why I want to make sure we isolate it to only those still running Rails 6.1 which eventually we'll drop support for.

Reply:
I couldn't find any examples of checking the rails version in the ActiveAdmin codebase. Would you prefer feature detection or a version check?

For example, something like this?

```ruby
# Rails.version is a String, and both "6.1" <=> "7" and "7.1" <=> "7" work as expected
if Rails.version < "7"
```

or this?

```ruby
if tag.respond_to?(:attributes)
```

Reply:
@javierjulio I went with the latter approach, but am happy to revisit.

Reply:
I would use `if Rails::VERSION::MAJOR <= 6` to isolate the old method and that way the `else` condition is for the current approach. I would be ok with feature detection but favor the version check in this case because otherwise we'd be falling back to the old approach.

Reply:
Thanks! Feature detection didn't work anyway 🤦🏼‍♂️